### PR TITLE
BarbaraWidjono, Edges, MediaRanker-Revisted

### PR DESCRIPTION
**Pull request by BarbaraWidjono**
Unable to get my Works#Update controller test to pass. If you would be willing to comment on the syntax of it, that would be great. For the implementation that was provided to us, the users/index and users/show page would not load, so I changed the voting loops which now works.

### CODE REVIEW

File: test/controllers/works_controller_test.rb [hunk 1]
```diff
@@ -2,115 +2,280 @@
...
+#       work = {
+#         work: {
+#           title: new_work_title
+#         }
+#       }
+# # binding.pry
+#
+#       patch work_path(work_to_change.id), params[:work]
+# # binding.pry
+#       expect(work_to_change.title).must_equal new_work_title
```
Comment (droberts-sea) [inline test/controllers/works_controller_test.rb:217]: This is almost there. There are two things that are missing:

- When you add the `params` parameter to the `patch` method, you should do it as a keyword argument:
    ```ruby
    patch work_path(work_to_change.id), params: work
    ```
- You need to reload `work_to_change` from the database before the expectation on line 217: `work_to_change.reload`. `work_to_change` is a local variable that was loaded _before_ you made the change, which means if the DB is updated it won't reflect the change.

**Comment by droberts-sea** [comment]
# MediaRanker Revisted
## What We're Looking For

Feature | Feedback
---     | ---
**General** |
Appropriate Tests on WorksController | yes - would have liked to see `upvote` tested, since there's some interesting edge cases there.
Appropriate Tests on SessionsController | yes
Appropriate Tests on UsersController | yes - these could be better organized with `describe` blocks, but the tests themselves look good
Completed OAuth | yes
Basic Authorization (who can see what) | yes - good use of a controller filter
**Overall** | Good work overall!
